packages/capabilities/readme.md

```diff
@@ -25,3 +33,126 @@ import * as Voucher from '@web3-storage/capabilities/voucher'
+        }>, unknown>;
+        origin: Store.Schema.Schema<...>;
+    }>;
+}, CapabilityMatch<...> | DerivedMatch<...>>>
+```
+
+While this is a fairly complex type signature, most of the types exist to support the mechanics of the capability parser and can generally be ignored when using the capabilities. The most interesting part as a user is the definition in the `DerivedMatch` type constraint, which shows the inferred ability and the constraints upon the resource URI and the caveats. In the example above, the `can` field shows that this capability's ability is `"store/add"`, its resource URI (the `with` field) must have the `"did:"` scheme, and there are three caveats defined in the `nb` field: `link`, `size`, and `origin`, each of which have constraints on their allowed values.
+
+### Using the exported capabilities
+
+The capability object exposes three methods via the `TheCapabilityParser` interface: `create`, `invoke`, and `delegate`. 
+
+#### `create`
+
+The `create` method returns a "materialized" capability object, which is to say, a JS object containing the `can`, `with`, and `nb` fields needed to fully specify a UCAN capability.
+
+You must provide an input object containing a `with` resource URI that matches the constraints in the capability definition, as well as an `nb` object containing any caveats you want to include. If a capability has no caveats defined, or if all the caveats are optional, you may omit the `nb` field from the input.
+
+```ts
+const cap = Store.add.create({
+  with: 'did:key:z6MkwFPNubhwM66HNKeJYtBu1Rv9n1LZdJhbyhLFg97Qr6FG',
+  nb: {
+    link: 'bagbaieraspawtgooy5lptr7loyd3fxjsrgkamre3y6au3ga4df5bkhrxdkmq',
+    size: 20,
+  }
+})
+```
+
+The above would result in an object similar to the following:
+
+```js
+{
+  can: 'store/add',
+  with: 'did:key:z6MkwFPNubhwM66HNKeJYtBu1Rv9n1LZdJhbyhLFg97Qr6FG',
+  nb: {
+    link: 'bagbaieraspawtgooy5lptr7loyd3fxjsrgkamre3y6au3ga4df5bkhrxdkmq',
+    size: 20,
+  }
+}
+```
+
+#### `invoke`
+
+The `invoke` method returns an "invocation" of the capability, which can be executed against a ucanto service.
+
+Like `create`, `invoke` accepts `with` and `nb` fields, and the inputs must match the constraints in the capability definition.
+
+Because invocations are a type of UCAN, you also need to supply some UCAN-related options. At minimum, you need to include the `issuer`, which is a `Signer` capable of signing a UCAN, and `audience`, which identifies the recipient by DID. You can also include any of the optional fields in the interface definition below:
+
+```ts
+interface UCANOptions {
+  audience: Principal
+  lifetimeInSeconds?: number
+  expiration?: UCAN.UTCUnixTimestamp
+  notBefore?: UCAN.UTCUnixTimestamp
+
+  nonce?: UCAN.Nonce
+
+  facts?: Fact[]
+  proofs?: Proof[]
+}
+```
+
+In the example below, we assume that the `loadSigningKey` function returns a valid `Signer` object.
+
+```ts
+import * as DID from '@ipld/dag-ucan/did'
+
+const issuer = loadSigningKey()
```

Comment:
IMO it could be illustrative to show a working version of this
```
import * as ed25519 from "@ucanto/principal/ed25519";
const issuer = await ed25519.generate();
```